scylla/src/response/query_result.rs

```diff
@@ -444,6 +458,1180 @@ impl QueryRowsResult {
+            self.settings.colors.boolean,
+            "{}",
+            self.value
+        )
+    }
+}
+
+// Uuid
+impl fmt::Display for WrapperDisplay<'_, Uuid> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write_colored!(
+            f,
+            self.settings.print_in_color,
+            self.settings.colors.uuid,
+            "{}",
+            self.value
+        )
+    }
+}
+
+fn display_in_quotes_and_colored<'a>(
+    f: &mut fmt::Formatter<'_>,
+    value: &(dyn StringConvertible<'a> + 'a),
+    settings: &RowsDisplayerSettings,
+) -> fmt::Result {
+    write_colored!(f, settings.print_in_color, settings.colors.text, "'")?;
+    write_colored!(
+        f,
+        settings.print_in_color,
+        settings.colors.text,
+        "{}",
+        value
+    )?;
+    write_colored!(f, settings.print_in_color, settings.colors.text, "'")
+}
+
+// list or set
+impl fmt::Display for WrapperDisplay<'_, Vec<CqlValue>> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        // choose opening bracket
+        if let CqlValue::List(_) = self.cql_value {
+            write_colored!(
+                f,
+                self.settings.print_in_color,
+                self.settings.colors.collection,
+                "["
+            )?;
+        } else {
+            // it is a set
+            write_colored!(
+                f,
+                self.settings.print_in_color,
+                self.settings.colors.collection,
+                "{{"
+            )?;
+        }
+
+        // print elements
+        let mut first_ture = true;
+        let value: &Vec<CqlValue> = self.value;
+        for item in value {
+            let item_wrapper = get_item_wrapper(item, self.settings);
+
+            // write comma
+            if !first_ture {
+                write_colored!(
+                    f,
+                    self.settings.print_in_color,
+                    self.settings.colors.collection,
+                    ", "
+                )?;
+            }
+            first_ture = false;
+
+            // if item is text, display it in colored quotes, otherwise display it as is (e.g. number)
+            // to not confuse with text
+            if let CqlValue::Text(_item) = item {
+                display_in_quotes_and_colored(f, &*item_wrapper, self.settings)?;
+            } else {
+                write!(f, "{}", item_wrapper)?;
+            }
+        }
+
+        // choose closing bracket
+        if let CqlValue::List(_) = self.cql_value {
+            write_colored!(
+                f,
+                self.settings.print_in_color,
+                self.settings.colors.collection,
+                "]"
+            )
+        } else {
+            // it is a set
+            write_colored!(
+                f,
+                self.settings.print_in_color,
+                self.settings.colors.collection,
+                "}}"
+            )
+        }
+    }
+}
```

Comment:
:wrench: ditto: let's explicitly check that it's indeed a set, and use `matches!`.